flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -357,7 +357,8 @@ public void close() {
                                                         artifactFilePath.getName()));
                                         filesToUpload.add(
                                                 new FileUpload(
-                                                        Paths.get(artifacts.getValue().filePath),
+                                                        Paths.get(
+                                                                artifactFilePath.toUri().getPath()),
```

Comment:
```suggestion
                                                                artifactFilePath.toUri()),
```
This seems redundant given that we do `Paths.get`?

Reply:
If the `URI` of the artifactFilePath doesn't have a scheme, the `Paths.get` throws an `IllegalArgumentException` exception. In order to support both with and without schema, we have to use the `Paths.get` that takes a file path string.

By looking at the code, I notice that `artifactFilePath.toUri().getPath()` is the same as `artifactFilePath.getPath()`. I think we can use `artifactFilePath.getPath()`. What do you think?

Reply:
Ah, URI#getPath returns a string.

> we can use artifactFilePath.getPath()

sounds good 👍

Reply:
It is updated at 70ceb32 . Could you take another look?